plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4MinioDataLake.java

```diff
@@ -0,0 +1,176 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.containers;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.base.util.AutoCloseableCloser;
+import io.trino.testing.containers.Minio;
+import io.trino.testing.minio.MinioClient;
+import org.testcontainers.containers.Network;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.trino.plugin.hive.containers.Hive4HiveServer.HIVE_SERVER_PORT;
+import static io.trino.testing.containers.Minio.MINIO_ACCESS_KEY;
+import static io.trino.testing.containers.Minio.MINIO_SECRET_KEY;
+import static io.trino.testing.containers.TestContainers.getPathFromClassPathResource;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.Network.newNetwork;
+
+public class Hive4MinioDataLake
+        implements AutoCloseable
+{
+    /**
+     * In S3 this region is implicitly the default one. In Minio, however,
+     * if we set an empty region, it will accept any.
+     * So setting it by default to `us-east-1` simulates S3 better
+     */
+    public static final String MINIO_DEFAULT_REGION = "us-east-1";
+
+    private final String bucketName;
+    private final Minio minio;
+    private final Hive4HiveServer hiveServer;
+    private final Hive4Metastore hiveMetastore;
+    private final AutoCloseableCloser closer = AutoCloseableCloser.create();
+
+    private final Network network;
+
+    private State state = State.INITIAL;
+    private MinioClient minioClient;
+
+    public Hive4MinioDataLake(String bucketName)
+    {
+        this(bucketName, Hive4Metastore.HIVE4_IMAGE);
+    }
+
+    public Hive4MinioDataLake(String bucketName, String hiveImage)
+    {
+        this(bucketName, ImmutableMap.of("/opt/hive/conf/hive-site.xml", getPathFromClassPathResource("hive_minio_datalake/hive4-hive-site.xml")), hiveImage);
+    }
+
+    public Hive4MinioDataLake(String bucketName, Map<String, String> hiveFilesToMount, String hiveImage)
+    {
+        this.bucketName = requireNonNull(bucketName, "bucketName is null");
+        network = closer.register(newNetwork());
+        this.minio = closer.register(
+                Minio.builder()
+                        .withNetwork(network)
+                        .withEnvVars(ImmutableMap.<String, String>builder()
+                                .put("MINIO_ACCESS_KEY", MINIO_ACCESS_KEY)
+                                .put("MINIO_SECRET_KEY", MINIO_SECRET_KEY)
+                                .put("MINIO_REGION", MINIO_DEFAULT_REGION)
+                                .buildOrThrow())
+                        .build());
+
+        Hive4Metastore.Builder metastorebuilder = Hive4Metastore.builder()
```

Comment:
would be worth referencing the hive readme (use the tag version) containing the docker commands used to start the hive metastore and hive server

Reply:
ya thats good idea